src/pages/groupComparison/vennDiagram/VennUtils.ts

```diff
@@ -0,0 +1,391 @@
+        return rectangles[0];
+    } else if (rectangles.length === 2) {
+        const rectA = rectangles[0];
+        const rectB = rectangles[1];
+        // x direction
+        const xMin = Math.max(rectA.x, rectB.x);
+        const xMax = Math.min(rectA.x + rectA.xLength, rectB.x + rectB.xLength);
+        // y direction
+        const yMin = Math.max(rectA.y, rectB.y);
+        const yMax = Math.min(rectA.y + rectA.yLength, rectB.y + rectB.yLength);
+
+        if (xMin >= xMax || yMin >= yMax) {
+            // no intersection
+            return {
+                x: 0,
+                y: 0,
+                xLength: 0,
+                yLength: 0
+            };
+        } else {
+            return {
+                x: xMin,
+                y: yMin,
+                xLength: xMax - xMin,
+                yLength: yMax - yMin
+            };
+        }
+    } else {
+        return rectangleIntersection(
+            rectangleIntersection(
+                ...rectangles.slice(1)
+            ),
+            rectangles[0]
+        );
+    }
+}
+
+export function getRegionArea(
+    sets:string[],
+    setRectangles:SetRectangles
+) {
+    // start with the intersection size
+    let size = rectangleArea(rectangleIntersection(...sets.map(s=>setRectangles[s])));
+
+    // if there are other sets not included in the region, subtract them from the set
+    const numSets = Object.keys(setRectangles).length;
+    if (sets.length !== numSets) {
+        // since max 3 sets total, we know (area.sets, numSets) is either (1,3), (2,3) or (1,2)
+        if (sets.length === 2) {
+            // numSets is 3
+            // so just subtract the intersection of all 3
+            size -= rectangleArea(rectangleIntersection(..._.values(setRectangles)));
+        } else {
+            // area.sets.length is 1
+            if (numSets === 2) {
+                // just subtract the intersection
+                size -= rectangleArea(rectangleIntersection(..._.values(setRectangles)));
+            } else {
+                // numSets is 3
+                // so we need to subtract inclusion/exclusion style
+                // WLOG lets call our set A, and theres B and C
+                // then area(A) = size(A) - size(A and B) - size(A and C) + size(A and B and C)
+                const others = Object.keys(setRectangles).filter(s=> sets[0] !== s);
+                const rectA = setRectangles[sets[0]];
+                const rectB = setRectangles[others[0]];
+                const rectC = setRectangles[others[1]];
+                size -= rectangleArea(rectangleIntersection(rectA, rectB));
+                size -= rectangleArea(rectangleIntersection(rectA, rectC));
+                size += rectangleArea(rectangleIntersection(..._.values(setRectangles)));
+            }
+        }
+    }
+
+    return size;
+}
+
+function isPointInsideRectangle(x:number, y:number, rectangle:Rectangle) {
+    return ((x > rectangle.x) &&
```

Comment:
should it not be >= ?

Reply:
doesnt really matter for all our purposes